jdbc/src/main/java/org/postgis/DriverWrapper.java

```diff
@@ -269,68 +283,88 @@ public final void addGT(java.sql.Connection conn, boolean lw) throws SQLExceptio
 
-        public void addBinaryGeometries(Connection conn) throws SQLException {
+        /** {@inheritDoc} */
+        @Override
+        public void addBinaryGeometries(final Connection conn) throws SQLException {
             PGConnection pgconn = (PGConnection) conn;
             pgconn.addDataType("geometry", org.postgis.PGgeometryLW.class);
-            pgconn.addDataType("geography", org.postgis.PGgeometryLW.class);
+            pgconn.addDataType("geography", org.postgis.PGgeographyLW.class);
         }
     }
 
+
     /** addGISTypes for V7.2 pgjdbc */
     protected static class TypesAdder72 extends TypesAdder {
-        public void addGeometries(Connection conn) throws SQLException {
+
+        /** {@inheritDoc} */
+        @Override
+        public void addGeometries(final Connection conn) throws SQLException {
             org.postgresql.PGConnection pgconn = (org.postgresql.PGConnection) conn;
             pgconn.addDataType("geometry", org.postgis.PGgeometry.class);
             pgconn.addDataType("public.geometry", org.postgis.PGgeometry.class);
             pgconn.addDataType("\"public\".\"geometry\"", org.postgis.PGgeometry.class);
 
-            pgconn.addDataType("geography", org.postgis.PGgeometry.class);
-            pgconn.addDataType("public.geography", org.postgis.PGgeometry.class);
-            pgconn.addDataType("\"public\".\"geography\"", org.postgis.PGgeometry.class);
+            pgconn.addDataType("geography", org.postgis.PGgeography.class);
+            pgconn.addDataType("public.geography", org.postgis.PGgeography.class);
+            pgconn.addDataType("\"public\".\"geography\"", org.postgis.PGgeography.class);
         }
 
-		public void addBoxen(Connection conn) throws SQLException {
+        /** {@inheritDoc} */
+        @Override
+		public void addBoxen(final Connection conn) throws SQLException {
             org.postgresql.PGConnection pgconn = (org.postgresql.PGConnection) conn;
             pgconn.addDataType("box3d", org.postgis.PGbox3d.class);
             pgconn.addDataType("box2d", org.postgis.PGbox2d.class);
         }
 
-        public void addBinaryGeometries(Connection conn) throws SQLException {
+        /** {@inheritDoc} */
+        @Override
+        public void addBinaryGeometries(final Connection conn) throws SQLException {
             org.postgresql.PGConnection pgconn = (org.postgresql.PGConnection) conn;
             pgconn.addDataType("geometry", org.postgis.PGgeometryLW.class);
-            pgconn.addDataType("geography", org.postgis.PGgeometryLW.class);
+            pgconn.addDataType("geography", org.postgis.PGgeographyLW.class);
         }
     }
 
+
     /** addGISTypes for V8.0 (and hopefully newer) pgjdbc */
     protected static class TypesAdder80 extends TypesAdder {
-        public void addGeometries(Connection conn) throws SQLException {
+
+        /** {@inheritDoc} */
+        @Override
+        public void addGeometries(final Connection conn) throws SQLException {
```

Comment:
Is there a reason that geography is mapped to `PGgeometry` instead of `PGgeography` in this method, unlike the other methods?

Reply:
I don't immediately see a reason why.  It looks like an oversight in the refactoring.
It also looks like the mappings might be missing a mapping for geography that is not qualified by a schema.